crates/compiler/mono/src/ir.rs

```diff
@@ -9268,9 +9268,13 @@ fn assign_num_literal_expr<'a>(
     num_value: IntOrFloatValue,
     hole: &'a Stmt<'a>,
 ) -> Stmt<'a> {
-    let layout = layout_cache
-        .from_var(env.arena, variable, env.subs)
-        .unwrap();
+    let layout = match layout_cache.from_var(env.arena, variable, env.subs) {
```

Comment:
When there's an type error `from_var` returns `Err(Erroneous)`. I added to handle that case.